nqeit-iceberg-dremio-extension/src/main/java/org/projectnessie/integtests/dremio/IcebergDremioExtension.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.integtests.dremio;
+
+import java.util.Objects;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+
+public class IcebergDremioExtension implements ParameterResolver {
+
+  private static String readRequiredSystemProperty(String s) {
+    return Objects.requireNonNull(System.getProperty(s), "required system property not set: " + s);
```

Comment:
i'd argue it is not ugly because the message is clearly describing what system property is missing.
we implemented the skipping as `Assumptions.assumeTrue(System.getProperty("dremio.url") != null);`  using the junit mechanism
but once that property is set, we want the tests to fail hard when the other properties are missing (instead of silently skipping the test).
this is why we use requireNonNull here.